.github/workflows/_bazel-build-test.yml

```diff
@@ -26,7 +26,7 @@ jobs:
   build-and-test:
     # Don't run on forked repos.
     if: github.repository_owner == 'pytorch'
-    runs-on: [self-hosted, linux.4xlarge]
+    runs-on: [self-hosted, linux.4xlarge.nvidia.gpu]
```

Comment:
I'm a bit confuse about the other https://github.com/pytorch/pytorch/pull/95795 to have cpu support, is that what we current have already?  IMO, let's just keep it in this PR

On the other hand, your approach of creating one job for cpu while other for gpu makes sense to me.  So pull workflow would look as follows:

I suggest to use `test-matrix` here to keep the same syntax as other jobs

```
linux-bionic-cpu-py3_10-gcc7-bazel-test:
  name: linux-bionic-cpu-py3.10-gcc7-bazel-test
  uses: ./.github/workflows/_bazel-build-test.yml
  with:
    build-environment: linux-bionic-cuda11.7-py3.10-gcc7-bazel-test
    docker-image-name: pytorch-linux-bionic-cuda11.7-cudnn8-py3-gcc7
    cuda-version: cpu
    test-matrix: |
      { include: [
        { config: "default", shard: 1, num_shards: 1, runner: "linux.4xlarge" },
      ]}

linux-bionic-cuda11_7-py3_10-gcc7-bazel-test:
  name: linux-bionic-cuda11.7-py3.10-gcc7-bazel-test
  uses: ./.github/workflows/_bazel-build-test.yml
  with:
    build-environment: linux-bionic-cuda11.7-py3.10-gcc7-bazel-test
    docker-image-name: pytorch-linux-bionic-cuda11.7-cudnn8-py3-gcc7
    test-matrix: |
      { include: [
        { config: "default", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
      ]}
```